File: Level 1/Week01/Day1/kjh2.java
import java.util.*;

class Solution {
  public int[] solution(String s) {
    int[] alphabetLastIdx = new int['z' - 'a' + 1];
    Arrays.fill(alphabetLastIdx, -1);

    List<Integer> answer = new ArrayList<>();
    for (int i = 0; i < s.length(); i++) {
      char ch = s.charAt(i);
      int lastIdx = alphabetLastIdx[ch - 'a'];
      if (lastIdx == -1) {
        answer.add(-1);
      } else {
        answer.add(i - lastIdx);
      }
      alphabetLastIdx[ch - 'a'] = i;
    }

    return answer.stream().mapToInt(Integer::intValue).toArray();
  }
}

File: Level 1/Week02/Day11/kjh2.kt
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        List<Integer> removedDup = new ArrayList<>();
        for (int element : arr) {
            if (removedDup.size() > 0 && element == removedDup.get(removedDup.size() - 1)) {
                continue;
            }
            removedDup.add(element);
        }

        return removedDup.stream().mapToInt(i -> i).toArray();
    }
}
